File: Nth node from end of linked list.cpp
class Solution{
public:
    int getNthFromLast(struct Node *head, int n)
{
    if(head==NULL)
    return -1;
    struct Node *temp=head;
    int cnt=0;
    while(temp!=NULL)
    {
        temp=temp->next;
        cnt++;
    }
    if(n>cnt)
    return -1;
    int pos=cnt-n;
    temp=head;
    while(pos)
    {
        temp=temp->next;
        pos--;
    }
    return temp->data;
}
};

File: Nth node from end of linked list.java
class Solution
{
    //Function to find the data of nth node from the end of a linked list.
    int getNthFromLast(Node head, int n)
    {
    	// Your code here
    	int len=0;
    	Node temp=head;
    	while(temp!=null){
    	    len++;
    	    temp=temp.next;
    	}
    	if(n>len)return -1;
    	
        temp=head;
        int ind=1;
        while(ind<=len && ind!=len-(n-1)){
            temp=temp.next;
            ind++;
        }
    	return temp.data;
    }
}
